mender-client.sh

```diff
@@ -187,8 +187,12 @@ function check_deployment() {
     elif [ "$status_code" == "401" ]; then
       log "JWT token expired, obtain a new one"
       JWT=$(get_jwt)
+    elif [ "$status_code" == "204" ]; then
+      echo -n "."
+    else
+      log "WARNING: Unexpected HTTP Status $status_code"
```

Comment:
If the unexpected status code should not break the loop also?

Reply:
It was not breaking so I am not confident on changing that.

The motivation for this change is to catch for example if we are being rate limited by the server, but we still want to continue until we get the update.